internal/server/store.go

```diff
@@ -828,15 +830,29 @@ func (s *Store) GetRelatedAtTime(uri string, predicate string, inverse bool, tar
 					continue
 				}
 
+				// get recorded time on relationship
+				// skip over all entries until et gt than query time
+				et := int64(binary.BigEndian.Uint64(k[18:]))
```

Comment:
if queryTime could be of type uint64, we would not need the cast here and in l869